common/core/desktop/tests/unit/kmnkbd/action_items.hpp

```diff
@@ -0,0 +1,101 @@
+#include <keyman/keyboardprocessor.h>
+#include <string>
+#include <iostream>
+
+#include "../test_assert.h"
+
+namespace
+{
+
+inline
+bool operator==(km_kbp_option_item const & lhs, km_kbp_option_item const & rhs)
+{
+  return lhs.scope == rhs.scope
+      && std::u16string(lhs.key) == rhs.key
+      && std::u16string(lhs.value) == rhs.value;
+}
+
+const char *action_item_types[] = {
+  "KM_KBP_IT_END", //         = 0,  // Marks end of action items list.
+  "KM_KBP_IT_CHAR", //        = 1,  // A Unicode character has been generated.
+  "KM_KBP_IT_MARKER", //      = 2,  // Correlates to kmn's "deadkey" markers.
+  "KM_KBP_IT_ALERT", //       = 3,  // The keyboard has triggered a alert/beep/bell.
+  "KM_KBP_IT_BACK", //        = 4,  // Delete the codepoint preceding the insertion point.
+  "KM_KBP_IT_PERSIST_OPT", // = 5,  // The indicated option needs to be stored.
+  "KM_KBP_IT_EMIT_KEYSTROKE", // = 6,  // Emit the current keystroke to the application
+  "KM_KBP_IT_INVALIDATE_CONTEXT", // = 7,
```

Comment:
Depending on what's get merged first, this might have to be extended for `KM_KBP_IT_CAPSLOCK`.